### ISSUE TITLE
[ISAAC-497] Implement an enhanced RAG Pipeline for Scientific/Research Workflows

### ISSUE THREAD

**Issue by eimenhmdt** [aietal/isaac#45]
Our current RAG (Retriever-Augmented Generation) pipeline, primarily designed for generic tasks, is facing significant challenges, particularly in its application to scientific and research workflows. There are notable issues in its stability and effectiveness, necessitating a complete overhaul.

A potential direction for this redesign is the integration of a framework like Llama Index, which could provide advanced capabilities in document retrieval and processing. However, the main task at hand is to fundamentally rethink our approach to constructing such a pipeline, especially tailored for scientific contexts.

Key considerations for this redesign include:

1. **Document Management**: We currently have a mix of uploaded documents and saved references from Semantic Scholar. A major part of this redesign will involve devising a strategy to unify these diverse sources into a cohesive system, ensuring seamless access and processing.
2. **AI Accessibility**: The pipeline should allow the AI to access the user's documents directly. This implies building robust and secure pathways for AI-user document interaction.
3. **Performance Optimization**: Speed is crucial. The new pipeline should be optimized for rapid retrieval and processing without compromising on accuracy or reliability.
4. **Citation and Referencing**: The AI must be capable of properly citing answers, drawing from both the user's documents and external references. This requires an intelligent and context-aware citation mechanism.
5. **Integration of Llama Index**: Explore the feasibility and benefits of integrating Llama Index or similar frameworks. This could potentially enhance the pipeline's capabilities in handling complex scientific data and queries.

This project is critical for advancing our AI's ability to interact with and process scientific documents effectively. We are looking for contributors with expertise in AI, RAG, and document management systems, particularly those who have a keen interest in applying these technologies in scientific research contexts. The goal is to build a RAG pipeline that is not only bug-free and stable but also sophisticated in its handling of scientific data and user interaction.

<sub>From [SyncLinear.com](https://synclinear.com) | [ISAAC-497](https://linear.app/isaac-editor/issue/ISAAC-497/implement-an-enhanced-rag-pipeline-for-scientificresearch-workflows)</sub>

**Comment by feliciien** [comment]
Hello, when I try to log in to the application and the app locally the page keeps refreshing

https://github.com/aietal/isaac/assets/116520199/fdf88a08-529a-4b42-9bec-02d4d691c7cb

.

**Comment by crediblebilal** [comment]
hey @feliciien  try removing the **refresh=true** param in the url.

**Comment by feliciien** [comment]
this one ?
<img width="498" alt="Screenshot 2023-12-19 at 4 49 06 PM" src="https://github.com/aietal/isaac/assets/116520199/723c81c1-3cce-45a5-98be-41d512413648">

**Comment by crediblebilal** [comment]
This will also work, but not sure if it might be useful in some other case.
Here, I am saying in your browser, instead of giving url as **localhost:3000/signup?refresh=true** just use **localhost:3000/signup**

**Comment by feliciien** [comment]
Yes, that is what I do but when I click on sign with email the URL change to "http://localhost:3000/signup?refresh=true" and keep refreshing even if I delete this part of the code. I can have access to the editor.

**Comment by polooner** [comment]
If anyone would like to collab, I am working on this!

**Comment by feliciien** [comment]
Hello @polooner I'm open to a collaboration I think we are having the same issue impossible to log in the app keeps refreshing.
can you please send me your Discord username? so we can get in touch.

**Comment by Player256** [comment]
Hello @feliciien and @polooner, I am interested on working this issue. Here's my discord: **best_space_assassin**. I may be available after 6th Jan. Let's do a collab.

**Comment by feliciien** [comment]
I send you a friend request

Diouf

**Comment by feliciien** [comment]
Ajoute-moi comme ami sur Discord ! L'invitation expire dans 1 semaine : https://discord.gg/EAEYRgYd

**Comment by feliciien** [comment]
I solved the refreshtrue issue but the editor page looks like this 

https://github.com/aietal/isaac/assets/116520199/512a966a-c3c6-4e46-a9c3-e1d2975b9119

**Comment by feliciien** [comment]
Never create a project ? It's don't look like the official website ? Missing updates ?

**Comment by Player256** [comment]
Hey how did you solve the signin issue?I still can't get into editor section

**Comment by feliciien** [comment]
You have to add the database manually(username…. ) you can send me a
message in my discord if you need more help

Diouf

**Comment by eimenhmdt** [comment]
Hey guys, sorry for being so unresponsive. I'm back now. 

To resolve the sign in issue, you need to create a database function called "create_profile_for_user" with the following definition:

```
begin
  insert into public.profile(id, email)
  values(new.id, new.email);
  
  return new;
end;
```

Lmk, whether this fix works.

**Comment by Player256** [comment]
Hey @eimenhmdt , I went with your suggestion still the same.  

https://github.com/aietal/isaac/assets/92082372/e131ce3e-32c5-4a4f-8e32-e353d2727ce1

**Comment by eimenhmdt** [comment]
Did you try creating a new account and using that for signing in?

**Comment by Player256** [comment]
Yes, that was the new account

**Comment by eimenhmdt** [comment]
Hmm. Try to set the function's Security setting to "Definer" instead of "Invoker".

**Comment by atulkrs** [comment]
Can you assign this to me?

**Comment by eimenhmdt** [comment]
> Can you assign this to me?

@atulkrs There are already people working on this.

**Comment by jacobbridges** [comment]
> We currently have a mix of uploaded documents and saved references from Semantic Scholar. A major part of this redesign will involve devising a strategy to unify these diverse sources into a cohesive system, ensuring seamless access and processing.

Is the plan to support "diverse sources" in the future? Or to support only uploaded documents and Semantic Scholar references?
